src/OpenTelemetry.ResourceDetectors.Container/ContainerResourceDetector.cs

```diff
@@ -132,45 +152,127 @@ private static string RemovePrefixAndSuffixIfNeeded(string input, int startIndex
         return input.Substring(startIndex, endIndex - startIndex);
     }
 
+    private static string? ExtractK8sContainerId()
+    {
+        try
+        {
+            var host = Environment.GetEnvironmentVariable(K8sServiceHostKey);
+            var port = Environment.GetEnvironmentVariable(K8sServicePortKey);
+            var @namespace = Environment.GetEnvironmentVariable(K8sNamespaceKey);
+            var hostname = Environment.GetEnvironmentVariable(K8sHostnameKey);
+            var containerName = Environment.GetEnvironmentVariable(K8sContainerNameKey);
+            var url = $"https://{host}:{port}/api/v1/namespaces/{@namespace}/pods/{hostname}";
```

Comment:
What is the guarantee that this will be a well-formed URL?

Reply:
Nothing I think but in this case an exception will be thrown, catched and logged using the EventSource.